code/datums/components/storage/storage.dm

```diff
@@ -521,19 +541,25 @@
 
 //Abuses the fact that lists are just references, or something like that.
 /datum/component/storage/proc/signal_return_inv(datum/source, list/interface, recursive = TRUE)
+	SIGNAL_HANDLER
+
 	if(!islist(interface))
 		return FALSE
 	interface |= return_inv(recursive)
 	return TRUE
 
 /datum/component/storage/proc/topic_handle(datum/source, user, href_list)
+	SIGNAL_HANDLER
+
 	if(href_list["show_valid_pocket_items"])
 		handle_show_valid_items(source, user)
 
 /datum/component/storage/proc/handle_show_valid_items(datum/source, user)
 	to_chat(user, "<span class='notice'>[source] can hold: [can_hold_description]</span>")
 
 /datum/component/storage/proc/mousedrop_onto(datum/source, atom/over_object, mob/M)
+	BLOCKING_SIGNAL_HANDLER
+
 	set waitfor = FALSE
 	. = COMPONENT_NO_MOUSEDROP
```

Comment:
I'm a bit confused about this one. This proc doesn't actually block anything and will always return the correct value due to ``set waitfor = FALSE``.
In effect, it's no different from how butchering does it, only that it doesn't break it up into two procs. on_item_attack from butchering is classified as SIGNAL_HANDLER.

Reply:
this honestly doesn't even make sense to set wait for anyway, because all signals are already setting waitfor to false.

Reply:
but this lack of understanding about how signals actually get called is at the root of all the damn issues in the first place.

Reply:
especially people who try to make the value returned from signals meaningful.

They were a mistake.

Reply:
So what's the fix here? Remove `set waitfor = FALSE` if it's being called asynchronously anyway?

Reply:
well it doesn't do anything, may as well remove it

Reply:
@Jared-Fogle I think the BLOCKING_SIGNAL_HANDLER is the  wrong name

it should be something more like 

SIGNAL_HANDLER_DOES_SLEEP, so we know
A) this is bad, and you have to pay careful attention to when you return a value
B) we can find it later to see if it needs fixing when we decide how to approach these

Reply:
Sounds good to me. Will do all that shortly.

Reply:
@optimumtact The waitfor here does actually do something.
Without it, this proc always return null

Reply:
![image](https://user-images.githubusercontent.com/47324920/89930950-cf5dbf80-dc0b-11ea-8c9f-6d94b373456c.png)

Reply:
> but this lack of understanding about how signals actually get called is at the root of all the damn issues in the first place.

Reply:
that explains another of the bugs.

Reply:
@Jared-Fogle looks like the waitfor isn't superflous, define should still be in place though.

Reply:
@optimumtact I can bring it back. Would it potentially be smart to have the define do `set waitfor = FALSE`, or would that cause issues somewhere else?

Reply:
would likely cause issues somewhere else where someone is inadvertently relying on the behaviour in the left hand side of that screenshot.

Reply:
errm right hand rather